apps/extension/src/shared/components/grpc-endpoint-form/use-grpc-endpoint-form.ts

```diff
@@ -138,6 +138,7 @@ export const useGrpcEndpointForm = (isOnboarding: boolean) => {
         setConfirmChangedChainIdPromise(undefined);
       }
 
+      await clearWalletCreationHeight(); // changing chain id means the wallet birthday is no longer valid
```

Comment:
manually tested across different chain Ids, and it properly clears the wallet height.